### PR TITLE
The simulavr gdb fix 

### PR DESCRIPTION
**Pull request by SrimantaBarua**
Reintroduced the infinite loop in `send_vcont`. Hopefully slightly better implemented so that you can break out of it.

### CODE REVIEW

File: shlr/gdb/src/packet.c [hunk 1]
```diff
@@ -159,7 +159,14 @@ int read_packet(libgdbr_t *g) {
 		}
 	}
 	g->data_len = 0;
-	while (r_socket_ready (g->sock, 0, READ_TIMEOUT) > 0) {
+	while (1) {
```
Comment (radare) [inline shlr/gdb/src/packet.c]: what about adding a counter and abort after N retries? infinite loops are scary

Comment (SrimantaBarua) [reply shlr/gdb/src/packet.c]: Okay, but the counter won't work for vcont packets. Because the target might be waiting for IO.

**Comment by radare** [comment]
thats why it was hanging in mac :(

**Comment by SrimantaBarua** [comment]
Maybe signal handling behaviour is different in mac, compared to linux. I don't one to check on, could you test with latest commit?

**Comment by XVilka** [comment]
@SrimantaBarua can you please fix AppVeyor problem and rebase. @Maijin can you please check this on your Mac?

**Comment by radare** [comment]
tested on mac, and it seems to work

**Pull request merged**